src/components/Search/SearchResultSeries.js

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Button, Container, Icon, Segment } from 'semantic-ui-react';
+
+import { selectors as lessonsSelectors } from '../../redux/modules/lessons';
+import { selectors as mdbSelectors } from '../../redux/modules/mdb';
+import { selectors as tagSelectors } from '../../redux/modules/tags';
+import { isDebMode } from '../../helpers/url';
+import Link from '../Language/MultiLanguageLink';
+import ScoreDebug from './ScoreDebug';
+import SearchResultBase from './SearchResultBase';
+import { CT_LESSONS_SERIES } from '../../helpers/consts';
+import SearchResultCollection from './SearchResultCollection';
+
+class SearchResultSeries extends SearchResultBase {
+  state = {
+    showAll: false,
+  };
+
+  renderSerie = s => {
+    const { t }             = this.props;
+    const { logLinkParams } = this.buildCollectionLinkParams(s);
+
+    return (
+      <Button basic size="tiny" className="link_to_cu">
+        {s.name}
+        <Link
+          key={s.id}
+          onClick={() => this.logClick(...logLinkParams)}
+          to={`/lessons/series/c/${s.id}`}
+        >
+          <span className="margin-right-8 margin-left-8">
+            <Icon name="tasks" size="small" />
+            {`${t('search.showAll')} ${s.cuIDs.length} ${t('pages.collection.items.lessons-collection')}`}
+          </span>
+        </Link>
+      </Button>
+    );
+  };
+
+  buildCollectionLinkParams = c => {
+    const { queryResult: { search_result: { searchId } }, hit, rank, filters }      = this.props;
+    const { _index: index, _source: { mdb_uid: mdbUid, result_type: resultType }, } = hit;
+
+    return {
+      canonicalLinkParams: [{ id: c.id, content_type: 'CT_LECTURE_SERIES' }, this.getMediaLanguage(filters)],
+      logLinkParams: [mdbUid, index, resultType, rank, searchId]
+    };
+  };
+
+  getLowestLevelSeries = (series, rootId) => {
+    if (series.length <= 1)
+      return series[0];
+    const root = series.find(s => s.parent_id === rootId || (!s.parent_id && !rootId));
+    if (!(root?.children?.length !== 0)) {
+      return root;
+    }
+
+    return this.getLowestLevelSeries(series, root.id);
+  };
+
+  handleToggle = () => this.setState({ showAll: !this.state.showAll });
+
+  render() {
+    const {
+      t,
+      location,
+      hit,
+      getSerieBySource,
+      getSerieByTag,
+      nestedDenormCollectionWUnits,
+      getTagById,
+      wip: { lectures: wipL, series: wipS }
+    } = this.props;
+
+    const { showAll } = this.state;
+
+    if (wipL || wipS) {
+      return null;
+    }
+
+    const isByTag  = hit._type === 'lessons_series_by_tag';
+    const getSerie = isByTag ? getSerieByTag : getSerieBySource;
+
+    const { _score: score, _uid, _source: { mdb_uid } } = hit;
```

Comment:
Why do you need to use search internals?
This does not feels right.

Reply:
I don't understand, what is search internals?